File: parish360-core/src/main/java/org/parish360/core/usermanagement/dto/PermissionInfo.java
package org.parish360.core.usermanagement.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.parish360.core.common.enums.PermissionType;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class PermissionInfo {
    private String id;
    private String name;
    private String description;
    private String module;
    private PermissionType permissionType;
}

File: parish360_mobile/lib/features/users/domain/entities/permission_info.dart
class PermissionInfo {
  final String? id;
  final String? name;
  final String? description;
  final String? module;
  final String? permissionType;

  PermissionInfo({
    this.id,
    this.name,
    this.description,
    this.module,
    this.permissionType,
  });

  factory PermissionInfo.fromJson(Map<String, dynamic> json) {
    return PermissionInfo(
      id: json['id'] as String?,
      name: json['name'] as String?,
      description: json['description'] as String?,
      module: json['module'] as String?,
      permissionType: json['permission_type'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'module': module,
      'permission_type': permissionType,
    };
  }
}
